doc/sslang-lrm.tex

```diff
@@ -328,6 +233,84 @@ \subsection{User-defined operators}
 a look at Agda's user-defined operators.
 
 \section{Types}
+\noindent sslang's type system is composed of two fundamental components: type constructors and type application. All types in sslang are a combination of type constructors and applications. \\
+
+\noindent Type constructors may take zero or more type arguments that, when composed via type application, form a fully-qualified type. \\ 
+
+\noindent Constructors that take zero type arguments can be thought of as ``primitive types", such as integers, booleans, pure events (i.e. conditional variable), and unit (i.e. void). \\
+
+\noindent Constructors that take more than type argument can be thought of as ``higher-order types". Here's a summary of sslang's higher-order types:
+\begin{itemize}
+    \item \texttt{[<type>]}: An array containing values of type \texttt{<type>}
+    \item \texttt{(<type>, ...)}: A tuple containing values of type \texttt{<type>}
+    \item \texttt{<type1> $\rightarrow$ <type2>}: A function that takes a   \texttt{<type1>} and returns a  \texttt{<type2>}
+    \item \texttt{\&<type>}: A reference to a variable of type \texttt{<type>}
+    \item \texttt{Sched <type>}: A scheduled variable of type \texttt{<type>}. These variables are scheduled and managed by the SSM runtime and are used to synchronize routines
+\end{itemize}
+
+\noindent This all is summarized by the following grammar:
+\setlength{\grammarindent}{5em}
+\begin{grammar}
+<type> ::= (<type>)
+\alt [<type>]
+\alt (<type>(, <type>)+)
+\alt <type> $\rightarrow$ <type>
+\alt \&<type>
+\alt ()
+\alt <type-id> <type>*
+
+<type-id> ::= [A-Z][a-zA-Z0-9]* 
+\end{grammar}
+
+\noindent Note the generic definition of type constructors: simply one or more type identifiers. This implementation is meant to facilitate support for user-defined types and to provide a concise interface for sslang's type system to deal with.
+\section{Routines}
+\noindent Routines are defined at the top-level of a sslang file and consistent of the following components:
+\begin{itemize}
+    \item Routine name
+    \item Zero or more parameters, each of which are optionally typed
+    \item Optionally specified return type
+    \item Routine body
+\end{itemize}
+\noindent The compiler will perform type inference for untyped components and will verify that typed components are correct.
+
+\subsection{Routine Signatures}
+\noindent At the core of sslang routine signatures, we have two symbols that provide type information about a language construct. Their meanings are as follows:
+\begin{itemize}
+    \item \texttt{:} (Colon): ``is of type"
```

Comment:
Can you give an example of the two different forms upfront? 

In fact, I think the "two broad categories" could be stated earlier, since it outlines the top-down structure for the reader to understand the rest of the section.